src/app/services/search.service.ts

```diff
@@ -1,16 +1,23 @@
 import { SearchResult } from './../models/searchResult';
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
-import { Observable } from 'rxjs';
+import { Observable, BehaviorSubject } from 'rxjs';
 
 @Injectable({
   providedIn: 'root'
 })
 export class SearchService {
-  baseURL = 'https://cors-anywhere.herokuapp.com/http://api.deezer.com/search?';
-  params = 'limit=2&';
+  private baseURL = 'https://cors-anywhere.herokuapp.com/http://api.deezer.com/search?';
+  private params = '';
+  private dataSource = new BehaviorSubject<SearchResult>(new SearchResult());
+  data = this.dataSource.asObservable();
```

Comment:
Better to use next logic:

```
private _query: string;
private querySubject = Subject<string>();

set query(query: string) {
    this._query = query;
    this.querySubject.next(query);
}

get query(): string {
    return this._query;
}

get query$(): Obserable<string> {
    return this.querySubject.asObsevable();
}
```

And then in your header you can use

```
this.searchService.query = '<your_query>';
```

And then in your data component where you have data output you should have function that calls and process fetch from API. In this component you can simply subscribe to search service changes like so:

```
this.searchService.query$.subscribe(() => {
    this.getData(); // <- implement this fn
})
```